internal/commands/servergroup/modify.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/completion"
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/output"
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/resolver"
+	"github.com/UpCloudLtd/upcloud-go-api/v6/upcloud/request"
+	"github.com/spf13/pflag"
+)
+
+type modifyCommand struct {
+	*commands.BaseCommand
+	params modifyParams
+	resolver.CachingServerGroup
+	completion.ServerGroup
+}
+
+// ModifyCommand creates the "servergroup modify" command
+func ModifyCommand() commands.Command {
+	return &modifyCommand{
+		BaseCommand: commands.New(
+			"modify",
+			"Modify a server group",
+			"upctl servergroup modify 8abc8009-4325-4b23-4321-b1232cd81231 --title your-server-group",
+			"upctl servergroup modify my-server-group --anti-affinity-policy strict",
+			`upctl servergroup modify my-server-group --server my-server-1 --server my-server-2 --server my-server-3-`,
+			`upctl servergroup modify 8abc8009-4325-4b23-4321-b1232cd81231 --server 0bab98e5-b327-4ab8-ba16-738d4af7578b --server my-server-2`,
+			`upctl servergroup modify my-server-group --label env=dev`,
+		),
+	}
+}
+
+type modifyParams struct {
+	request.ModifyServerGroupRequest
+
+	antiAffinityPolicy string
+	labels             []string
+	servers            []string
+}
+
+var defaultModifyParams = modifyParams{
+	ModifyServerGroupRequest: request.ModifyServerGroupRequest{},
+}
+
+func (p *modifyParams) processParams(exec commands.Executor, uuid string) error {
+	p.UUID = uuid
+
+	if p.antiAffinityPolicy != "" {
+		p.AntiAffinityPolicy = upcloud.ServerGroupAntiAffinityPolicy(p.antiAffinityPolicy)
+	}
+
+	if len(p.labels) > 0 {
+		labelSlice, err := labels.StringsToUpCloudLabelSlice(p.labels)
+		if err != nil {
+			return err
+		}
+
+		p.Labels = labelSlice
+	}
+
+	if len(p.servers) > 0 {
+		servers, err := stringsToServerUUIDSlice(exec, p.servers)
+		if err != nil {
+			return err
+		}
+		p.Members = &servers
+	}
+
+	return nil
+}
+
+// InitCommand implements Command.InitCommand
+func (s *modifyCommand) InitCommand() {
+	fs := &pflag.FlagSet{}
+	fs.StringVar(&s.params.antiAffinityPolicy, "anti-affinity-policy", defaultModifyParams.antiAffinityPolicy, "Anti-affinity policy. Valid values are `yes` (best effort), `strict` and `no`. Will take effect upon server start.")
+	fs.StringArrayVar(&s.params.labels, "label", defaultModifyParams.labels, "Labels to describe the server in `key=value` format, multiple can be declared. If set, all the existing labels will be replaced with provided ones.\nUsage: --label env=dev\n\n--label owner=operations")
+	fs.StringVar(&s.params.Title, "title", defaultModifyParams.Title, "New server group title.")
+	fs.StringArrayVar(&s.params.servers, "server", defaultModifyParams.servers, "Servers that belong to the server group, multiple can be declared. If set, all the existing server entries will be replaced with provided ones.\nUsage: --server my-server\n\n--server 00333d1b-3a4a-4b75-820a-4a56d70395dd")
```

Comment:
Could have completion for `--server` argument similarly than there is in `servergroup create` command

Reply:
Added